.github/workflows/weekly.yml

```diff
@@ -0,0 +1,32 @@
+name: weekly
+
+on:
+  schedule:
+    # this script will run each Monday at 14:14
+    - cron: "14 14 * * 1"
+  # Uncomment below to test.
+  # push:
+  #   branches: [gha-test-*, canary, auto]
+
+jobs:
+  dep-report:
+    runs-on: ubuntu-latest
+    container:
+      image: diem/build_environment:main
+    env:
+      MESSAGE_PAYLOAD_FILE: /tmp/message
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+            # fetch full history
+            fetch-depth: 0
+      - uses: ./.github/actions/build-setup
+      - name: Install cargo-guppy
```

Comment:
If we added guppy to scripts/dev-setup.sh similar to sccache and grcov we wouldn't need to recompile it every time.   But since it's once a week I don't care (aside from the fact that it could break on a rust update).    Yeah... not good.   This tool should be in dev-setup.sh so that builds triggered by changes to the rust-version trigger the build of the guppy version we will us in this report.    Otherwise oncall will have more garbage to fix that should be addressed up front.    Since I'm fixing coverage reports now for just this reason.....

Reply:
I'm not sure if I'll be making changes in the scripts/dev-setup.sh for this task.
Can installing a custom rust toolchain here that works with cargo-guppy be a solution for this?
@bmwill, do you have any suggestion?

Reply:
@nasifimtiazohi Yeah lets not have you make changes to the dev-setup.sh script, I'll take an AI to make this change once this PR lands